src/query/service/src/interpreters/interpreter.rs

```diff
@@ -45,26 +49,45 @@ pub trait Interpreter: Sync + Send {
 
     /// The core of the databend processor which will execute the logical plan and get the DataBlock
     async fn execute(&self, ctx: Arc<QueryContext>) -> Result<SendableDataBlockStream> {
-        let mut build_res = self.execute2().await?;
+        log_query_start(&ctx);
+
+        let mut build_res = match self.execute2().await {
```

Comment:
`execute2` could be renamed into `execute_impl`

Reply:
It will be rename to execute in the other pr.